src/DotVVM.Framework/Hosting/DotvvmRequestContextExtensions.cs

```diff
@@ -232,7 +233,7 @@ public static void ReturnFile(this IDotvvmRequestContext context, Stream stream,
             AttachmentDispositionType = attachmentDispositionType ?? "attachment"
         };
 
-        var generatedFileId = returnedFileStorage.StoreFile(stream, metadata).Result;
+        var generatedFileId = returnedFileStorage.StoreFileAsync(stream, metadata).Result;
```

Comment:
Could we add async ReturnFile and deprecate this method? This does not seem right, maybe it's not very problematic, but at least there should be more correct way.

Reply:
@exyi  I have removed the suffix.

Reply:
@exyi What do you mean by the "deprecate this method"? When the signature of parameters is the same as before then we cannot add an overload with a different return type and deprecate it. So I am not sure what you mean.

Reply:
Ah, shit. In this case we'd need the async suffix... :(

Reply:
@exyi Ok, reverting the last commit and merging this PR.

Reply:
I don't understand how that solves this issue, there is still synchronous wait in this method